lib/helpers/query/castUpdate.js

```diff
@@ -41,7 +40,11 @@ module.exports = function castUpdate(schema, obj, options, context, filter) {
     }
     return obj;
   }
-
+  if (schema.options.strict === 'throw') {
```

Comment:
Still needs a check for whether `obj.hasOwnProperty(schema.options.discriminatorKey)`